qiskit_experiments/database_service/db_experiment_data.py

```diff
@@ -257,20 +289,57 @@ def add_data(
             if not self._service:
                 self._set_service_from_backend(self._backend)
 
-            self._jobs[job.job_id()] = job
-
-        if job_data:
-            job_kwargs = {
-                "jobs": job_data,
-                "timeout": timeout,
-            }
-            self._job_futures.append(
-                (job_kwargs, self._executor.submit(self._add_jobs_data, **job_kwargs))
-            )
+            self._jobs[jid] = job
+            self._add_job_future(job)
 
         if self.auto_save:
             self.save_metadata()
 
+    def _add_job_future(self, job):
+        """Submit new _add_job_data job to executor"""
+        jid = job.job_id()
+        if jid in self._job_futures:
+            LOG.warning("Job future has already been submitted [jid: %s]", jid)
+        else:
+            self._job_futures[jid] = self._job_executor.submit(self._add_job_data, job)
+
+    def _add_job_data(
+        self,
+        job: Union[Job, BaseJob],
+    ) -> Tuple[str, JobStatus]:
+        """Wait for a job to finish and add job result data.
+
+        Args:
+            job: the Job to wait for and add data from.
+
+        Returns:
+            A tuple of the job id and final job status.
+
+        Raises:
+            Exception: If an error occured when adding job data.
+        """
+        jid = job.job_id()
+        try:
+            job_result = job.result()
```

Comment:
Any timeout value can be passed here? I always set timeout especially for regular monitoring type jobs (i.e. Given I run T1 every hour. If other user interleaves a heavy job on the same backend, then my job will not be finished within hour. Then I want to skip that job and wait for next hour instead of stacking jobs. Timeout of job is very convenient to manage such behavior).

Reply:
This is an interesting point. The idea here was I wanted to remove timeout from adding jobs, since it was never actually used and since it is run in its own future it can in principle wait forever and we handle `timeout` for that future itself, which is done using block_for_results/analysis results` etc.

How do you do that with a T1 experiment though since you can never pass the timeout arg here through the experiment API doing `t1exp.run()`? Wouldn't you have to access jobs and then call that manually? 
In that case I think you could could do something like

```block
expdata = exp.run()
expdata.block_for_results(timeout=timeout)
if expdata.job_status() != JobStatus.Done:
    expdata.cancel()
```

If we keep timeout in `add_jobs` that would basically force the experiment to cancel if the jobs time out and we would need to add a timeout kwarg to `BaseExperiment.run` to be able to set it. But if an experiment is split into multiple jobs, that value would be used for each job rather than the whole experiment.